app/controllers/food_items_controller.rb

```diff
@@ -3,41 +3,47 @@ class FoodItemsController < ApplicationController
   before_action :set_food_item, only: [ :update, :destroy ]
   before_action :initialize_food_item_service, only: [ :create, :update, :destroy ]
 
-  # GET /food_items
   def index
     @food_items = FoodItemQuery.new(params: params).call
   end
 
-  # POST /food_items
   def create
     result = @food_item_service.create_item(food_item_params)
 
     if result[:success]
-      handle_result(result, dashboard_path)
+      handle_result(result, dashboard_menu_path)
     else
       @food_item = result[:food_item]
       @food_items = FoodItem.all
-      render "dashboard/index", status: :unprocessable_entity
+      render "dashboard/menu", status: :unprocessable_entity
     end
   end
 
-  # PATCH /food_items/:id
   def update
+    if params[:reactivate]
+      @food_item = FoodItem.unscoped.find(params[:id])
+      if @food_item.update(deleted_at: nil)
+        redirect_to dashboard_menu_path, notice: "#{@food_item.name} has been reactivated!"
+      else
+        redirect_to dashboard_menu_path, alert: "Failed to reactivate item."
+      end
+      return
+    end
```

Comment:
This feels like should be another action/endpoint, It doesn't share any of the logic from the rest of the method